README.md

```diff
@@ -1,24 +1,279 @@
 
-* Database initialization
+#### Custom Fields
+TODO
 
-* How to run the test suite
+## Installation
+TODO
 
-* Services (job queues, cache servers, search engines, etc.)
+## Services (job queues, cache servers, search engines, etc.)
+URI Services stores local, temporary and external terms in a mysql database. Solr is used to index terms for fast lookup and searching. Redis/Resque will be used to run scheduled jobs.
 
-* Deployment instructions
+## Deployment instructions
+TODO
 
-* ...
+## API v1
+
+### Vocabularies
+#### List Vocabularies
+- **Request**
+  `GET /vocabularies`
+
+- **Success Response**
+
+  Sorts vocabularies alphabetically by label. Defaults to `page = 1` and `per_page = 20`.
+
+  _Status:_ 200
+
+  _Body:_
+
+  ```json
+  {
+    "page": 1,
+    "per_page": 10,
+    "total_results": 2,
+    "vocabularies": [
+      { "string_key": "name" , "label": "Names" },
+      { "string_key": "collections" , "label": "Collections" }
+    ]
+  }
+  ```
+
+#### Find Single Vocabulary
+- **Request**
+
+  `GET /vocabularies/:string_key`
+
+- **Successful Response**
+
+  _Status:_ 200
+
+  _Body:_
+  ```json
+  {
+     "string_key": "names",
+     "label": "Names"
+  }
+  ```
+
+#### Create Vocabulary
+- **Request**
+
+  `POST /vocabularies`
+
+  `string_key` is a required parameter.
+
+- **Success Response**
+
+  _Status:_ `201`
+
+  _Body:_
+    ```json
+    {
+      "string_key": "names",
+      "label": "Names"
+    }
+    ```
+
+#### Update Vocabulary
+- **Request**
+
+  `PATCH /vocabularies/:string_key`
+
+- **Success Response**
+
+  _Status:_ 200
+
+  _Body:_
+  ```json
+  {
+     "string_key": "names",
+     "label": "Names"
+  }
+  ```
+
+#### Delete Vocabulary
+- **Request**
+
+  `DELETE /vocabularies/:string_key`
+
+- **Success Response**
+
+  _Status:_ 204
+
+  _Body:_ nil
+
+- **Error Response**
+  - If resource does not exists returns `404`
+  - If not successful return `500`
+
+
+#### Send back description/instruction
+- **Request**
+
+  `OPTIONS /vocabularies/`
+- **Success Response**
+  TODO
+
+### Vocabularies/Term
+
+#### List/Search through Terms in Vocabulary
+- **Request**
+
+  `GET /vocabularies/:string_key/terms`
+
+  Examples:
+  ```
+  GET /vocabularies/:string_key/terms?page=1&per_page=20`
+  GET /vocabularies/:string_key/terms?q=smith&authority=naf&page=1&per_page=20`
+  GET /vocabularies/:string_key/terms?uri=https://creativecommons.org/publicdomain/zero/1.0/
+  GET /vocabularies/:string_key/terms?value=Smith # returns temp term b/c no uri was given
+  ```
+
+  Allowed query fields:
+    - `q`: query uri terms
+    - `authority`: facet by authority
+    - `uri`: search for uri
+    - `value`: search for value with in temp terms
+
+  Other allowed params:
+    - `per_page`: number of terms per page, default: 20
+    - `page`: page to display
+
+- **Success Response**
+
+  Sorting alphabetically by ????
```

Comment:
```suggestion
  Sorted primarily by match score (in the case of a q search), secondarily by label alphabetical sort
```

Reply:
FYI: by "match score" I mean by whatever order solr gives us. on the solr side, we'll primarily sort by score, and the second sort comparison will be alphabetical label sort.

Reply:
gotcha